### PR TITLE
fix console range iteration edge case

### PR DESCRIPTION
**Pull request by rjullman**

### CODE REVIEW

**Comment by rjullman** [comment]
I want to write a test for this first... but this is a little bit annoying.  I may want to write a testing framework like for cli.

**Comment by rjullman** [comment]
Related to https://jira.yojoe.local/browse/QA-94641

**Comment by rjullman** [comment]
I don't want to add the additional testing infrastructure until I can use the same dagger stuff from clis, but that is blocked on having atlas be a 3rd party library for internal code (since we don't use preprocessors).  I'm currently setting up a stack to test this locally more fully, but I vote for pushing this through (testing this with the current constraints is tricky).

File: atlasdb-server/src/main/java/com/palantir/atlasdb/impl/AtlasDbServiceImpl.java [hunk 1]
```diff
@@ -137,12 +136,13 @@ public RangeToken execute(Transaction t) {
                     .build();
                 BatchingVisitable<RowResult<byte[]>> visitable = t.getRange(getTableRef(range.getTableName()), request);
                 List<RowResult<byte[]>> results = BatchingVisitables.limit(visitable, limit).immutableCopy();
-                TableRowResult data = new TableRowResult(range.getTableName(), results);
                 if (results.size() == limit) {
+                    TableRowResult data = new TableRowResult(range.getTableName(), results.subList(0, limit - 1));
                     RowResult<byte[]> lastResult = results.get(limit - 1);
```
Comment (clockfort) [inline atlasdb-server/src/main/java/com/palantir/atlasdb/impl/AtlasDbServiceImpl.java]: lastResultInBatch might be less confusing

**Comment by jboreiko** [comment]
Don't mean to be the guy commenting after merge, but it doesn't look like it would be too hard to mock the transaction manager to return a transaction that returns some set of results that hits this boundary condition.
